data/abilities.ts

```diff
@@ -365,6 +365,13 @@ export const Abilities: import('../sim/dex-abilities').AbilityDataTable = {
 				source.abilityState.battleBondTriggered = true;
 			}
 		},
+		onModifyMovePriority: -1,
+		onModifyMove(move, attacker) {
+			if (move.id === 'watershuriken' && attacker.species.name === 'Greninja-Ash' &&
+				!attacker.transformed) {
```

Comment:
In Gen 9, Greninja doesn't actually transform; see the `onSourceAfterFaint` code above. (The Gen 8 code should stay the same, of course.) Untested:
```suggestion
			if (move.id === 'watershuriken' && attacker.abilityState.battleBondTriggered) {
```

Reply:
This isn't for regular Gen 9 Greninja with Battle Bond—it's for Ash-Greninja in Hackmons.

Reply:
The current implementation is correct. Greninja's Water Shuriken bonus doesn't work if the user is Transformed.